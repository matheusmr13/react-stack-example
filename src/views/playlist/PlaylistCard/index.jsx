import React from 'react';

import './PlaylistCard.css';

const PlaylistCard = ({ playlist }) => {
	return (
		<div className="playlist-card">
			<img src={playlist.images[0].url} alt={playlist.name} />
			<div className="playlist-card__name">{playlist.name}</div>
		</div>
	);
};

export default PlaylistCard;
