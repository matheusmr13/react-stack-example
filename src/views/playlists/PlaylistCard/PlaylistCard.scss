.playlist-card {
	position: relative;
	display: inline-block;
	cursor: pointer;

	&__name {

	}

	&:hover {
		&::after {
			content: '';
			position: absolute;
			top: 0;
			left: 0;
			width: 100%;
			height: 100%;
			background-color: rgba(0, 0, 0, 0.3);
		}
	}
}