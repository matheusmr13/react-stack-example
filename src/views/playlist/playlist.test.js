import React from 'react';
import { shallow } from 'enzyme';
import PlaylistList from './index';
import PlaylistCard from './card';

const playlist = {
	external_urls: {
		spotify: 'mylink'
	},
	id: '6ftJBzU2LLQcaKefMi7ee7',
	images: [{
		url: 'myotherlink'
	}],
	name: 'my playlist',
	owner: {
		external_urls: {
			spotify: 'owner link'
		},
		id: 'ownerid'
	}
};
const playlists = [{ ...playlist }, {
	...playlist,
	name: 'dif name'
}];

describe('PlaylistCard component', () => {
	const fetchInitialPlaylists = jest.fn();
	const playlistList = shallow(<PlaylistList
		playlists={[]}
		fetchInitialPlaylists={fetchInitialPlaylists}
		loadingPlaylists
	/>);
	it('should render loading properly', () => {
		expect(playlistList).toMatchSnapshot();
	});
	it('should render list properly', () => {
		playlistList.setProps({
			playlists,
			fetchInitialPlaylists,
			loadingPlaylists: false
		});
		expect((playlistList).find(PlaylistCard)).toHaveLength(2);
		expect(fetchInitialPlaylists).toHaveBeenCalled();
		expect(playlistList).toMatchSnapshot();
	});
	it('should filter playlists by name', () => {
		playlistList.instance().onFilterByName({}, 'my');
		expect(playlistList.instance().state.filteredPlaylists).toHaveLength(1);
	});
});
