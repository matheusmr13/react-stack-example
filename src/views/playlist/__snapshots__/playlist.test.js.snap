// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistCard component should render list properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Playlists
    fetchInitialPlaylists={[Function]}
    loadingPlaylists={false}
    playlists={
        Array [
            Object {
              "external_urls": Object {
                "spotify": "mylink",
              },
              "id": "6ftJBzU2LLQcaKefMi7ee7",
              "images": Array [
                Object {
                  "url": "myotherlink",
                },
              ],
              "name": "my playlist",
              "owner": Object {
                "external_urls": Object {
                  "spotify": "owner link",
                },
                "id": "ownerid",
              },
            },
            Object {
              "external_urls": Object {
                "spotify": "mylink",
              },
              "id": "6ftJBzU2LLQcaKefMi7ee7",
              "images": Array [
                Object {
                  "url": "myotherlink",
                },
              ],
              "name": "dif name",
              "owner": Object {
                "external_urls": Object {
                  "spotify": "owner link",
                },
                "id": "ownerid",
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <TextField
                    disabled={false}
                    floatingLabelFixed={false}
                    fullWidth={false}
                    hintText="Filter..."
                    multiLine={false}
                    onChange={[Function]}
                    rows={1}
                    type="text"
                    underlineShow={true}
                    value=""
          />
</div>,
        <div>
          <PlaylistCard
                    playlist={
                              Object {
                                        "external_urls": Object {
                                          "spotify": "mylink",
                                        },
                                        "id": "6ftJBzU2LLQcaKefMi7ee7",
                                        "images": Array [
                                          Object {
                                            "url": "myotherlink",
                                          },
                                        ],
                                        "name": "my playlist",
                                        "owner": Object {
                                          "external_urls": Object {
                                            "spotify": "owner link",
                                          },
                                          "id": "ownerid",
                                        },
                                      }
                    }
          />
          <PlaylistCard
                    playlist={
                              Object {
                                        "external_urls": Object {
                                          "spotify": "mylink",
                                        },
                                        "id": "6ftJBzU2LLQcaKefMi7ee7",
                                        "images": Array [
                                          Object {
                                            "url": "myotherlink",
                                          },
                                        ],
                                        "name": "dif name",
                                        "owner": Object {
                                          "external_urls": Object {
                                            "spotify": "owner link",
                                          },
                                          "id": "ownerid",
                                        },
                                      }
                    }
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TextField
            disabled={false}
            floatingLabelFixed={false}
            fullWidth={false}
            hintText="Filter..."
            multiLine={false}
            onChange={[Function]}
            rows={1}
            type="text"
            underlineShow={true}
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "floatingLabelFixed": false,
            "fullWidth": false,
            "hintText": "Filter...",
            "multiLine": false,
            "onChange": [Function],
            "rows": 1,
            "type": "text",
            "underlineShow": true,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <PlaylistCard
              playlist={
                            Object {
                                          "external_urls": Object {
                                            "spotify": "mylink",
                                          },
                                          "id": "6ftJBzU2LLQcaKefMi7ee7",
                                          "images": Array [
                                            Object {
                                              "url": "myotherlink",
                                            },
                                          ],
                                          "name": "my playlist",
                                          "owner": Object {
                                            "external_urls": Object {
                                              "spotify": "owner link",
                                            },
                                            "id": "ownerid",
                                          },
                                        }
              }
/>,
            <PlaylistCard
              playlist={
                            Object {
                                          "external_urls": Object {
                                            "spotify": "mylink",
                                          },
                                          "id": "6ftJBzU2LLQcaKefMi7ee7",
                                          "images": Array [
                                            Object {
                                              "url": "myotherlink",
                                            },
                                          ],
                                          "name": "dif name",
                                          "owner": Object {
                                            "external_urls": Object {
                                              "spotify": "owner link",
                                            },
                                            "id": "ownerid",
                                          },
                                        }
              }
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "6ftJBzU2LLQcaKefMi7ee7",
            "nodeType": "function",
            "props": Object {
              "playlist": Object {
                "external_urls": Object {
                  "spotify": "mylink",
                },
                "id": "6ftJBzU2LLQcaKefMi7ee7",
                "images": Array [
                  Object {
                    "url": "myotherlink",
                  },
                ],
                "name": "my playlist",
                "owner": Object {
                  "external_urls": Object {
                    "spotify": "owner link",
                  },
                  "id": "ownerid",
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6ftJBzU2LLQcaKefMi7ee7",
            "nodeType": "function",
            "props": Object {
              "playlist": Object {
                "external_urls": Object {
                  "spotify": "mylink",
                },
                "id": "6ftJBzU2LLQcaKefMi7ee7",
                "images": Array [
                  Object {
                    "url": "myotherlink",
                  },
                ],
                "name": "dif name",
                "owner": Object {
                  "external_urls": Object {
                    "spotify": "owner link",
                  },
                  "id": "ownerid",
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <TextField
                        disabled={false}
                        floatingLabelFixed={false}
                        fullWidth={false}
                        hintText="Filter..."
                        multiLine={false}
                        onChange={[Function]}
                        rows={1}
                        type="text"
                        underlineShow={true}
                        value=""
            />
</div>,
          <div>
            <PlaylistCard
                        playlist={
                                    Object {
                                                "external_urls": Object {
                                                  "spotify": "mylink",
                                                },
                                                "id": "6ftJBzU2LLQcaKefMi7ee7",
                                                "images": Array [
                                                  Object {
                                                    "url": "myotherlink",
                                                  },
                                                ],
                                                "name": "my playlist",
                                                "owner": Object {
                                                  "external_urls": Object {
                                                    "spotify": "owner link",
                                                  },
                                                  "id": "ownerid",
                                                },
                                              }
                        }
            />
            <PlaylistCard
                        playlist={
                                    Object {
                                                "external_urls": Object {
                                                  "spotify": "mylink",
                                                },
                                                "id": "6ftJBzU2LLQcaKefMi7ee7",
                                                "images": Array [
                                                  Object {
                                                    "url": "myotherlink",
                                                  },
                                                ],
                                                "name": "dif name",
                                                "owner": Object {
                                                  "external_urls": Object {
                                                    "spotify": "owner link",
                                                  },
                                                  "id": "ownerid",
                                                },
                                              }
                        }
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TextField
              disabled={false}
              floatingLabelFixed={false}
              fullWidth={false}
              hintText="Filter..."
              multiLine={false}
              onChange={[Function]}
              rows={1}
              type="text"
              underlineShow={true}
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "floatingLabelFixed": false,
              "fullWidth": false,
              "hintText": "Filter...",
              "multiLine": false,
              "onChange": [Function],
              "rows": 1,
              "type": "text",
              "underlineShow": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <PlaylistCard
                playlist={
                                Object {
                                                "external_urls": Object {
                                                  "spotify": "mylink",
                                                },
                                                "id": "6ftJBzU2LLQcaKefMi7ee7",
                                                "images": Array [
                                                  Object {
                                                    "url": "myotherlink",
                                                  },
                                                ],
                                                "name": "my playlist",
                                                "owner": Object {
                                                  "external_urls": Object {
                                                    "spotify": "owner link",
                                                  },
                                                  "id": "ownerid",
                                                },
                                              }
                }
/>,
              <PlaylistCard
                playlist={
                                Object {
                                                "external_urls": Object {
                                                  "spotify": "mylink",
                                                },
                                                "id": "6ftJBzU2LLQcaKefMi7ee7",
                                                "images": Array [
                                                  Object {
                                                    "url": "myotherlink",
                                                  },
                                                ],
                                                "name": "dif name",
                                                "owner": Object {
                                                  "external_urls": Object {
                                                    "spotify": "owner link",
                                                  },
                                                  "id": "ownerid",
                                                },
                                              }
                }
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "6ftJBzU2LLQcaKefMi7ee7",
              "nodeType": "function",
              "props": Object {
                "playlist": Object {
                  "external_urls": Object {
                    "spotify": "mylink",
                  },
                  "id": "6ftJBzU2LLQcaKefMi7ee7",
                  "images": Array [
                    Object {
                      "url": "myotherlink",
                    },
                  ],
                  "name": "my playlist",
                  "owner": Object {
                    "external_urls": Object {
                      "spotify": "owner link",
                    },
                    "id": "ownerid",
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "6ftJBzU2LLQcaKefMi7ee7",
              "nodeType": "function",
              "props": Object {
                "playlist": Object {
                  "external_urls": Object {
                    "spotify": "mylink",
                  },
                  "id": "6ftJBzU2LLQcaKefMi7ee7",
                  "images": Array [
                    Object {
                      "url": "myotherlink",
                    },
                  ],
                  "name": "dif name",
                  "owner": Object {
                    "external_urls": Object {
                      "spotify": "owner link",
                    },
                    "id": "ownerid",
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PlaylistCard component should render loading properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Playlists
    fetchInitialPlaylists={[Function]}
    loadingPlaylists={true}
    playlists={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "Loading",
  Symbol(enzyme.__nodes__): Array [
    "Loading",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
